src/lib/message/L1Transaction.ts

```diff
@@ -239,11 +240,21 @@ export class L1TransactionReceipt implements TransactionReceipt {
       .map(m => {
         const value = this.parseEthDepositData(m.inboxMessageEvent.data)
 
+        // we need to apply an alias to the sender to get the to address, but only if the sender was an EoA
+        // https://github.com/OffchainLabs/nitro/blob/a4eb505ae4f47967fa0ec18e85a43aa4602540c5/contracts/src/bridge/Inbox.sol#L216
+        const aliasedSender = new Address(
+          m.bridgeMessageEvent.sender
+        ).applyAlias().value
+        const to =
+          this.from.toLowerCase() === aliasedSender.toLowerCase()
+            ? aliasedSender
+            : m.bridgeMessageEvent.sender
+
         return new EthDepositMessage(
           l2Provider,
           chainID,
           m.inboxMessageEvent.messageNum,
-          m.bridgeMessageEvent.sender,
+          to,
```

Comment:
I think we should refactor this to expect the `from` address (ie whoever called the L1 inbox.
We can then internally calculate the alias if needed (ie if we detect its a L1 contract)